Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusPublic.ts

```diff
@@ -0,0 +1,285 @@
+import { ActionType } from "@/constants/actionType";
+import { ResultType } from "@/constants/resulttype";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import CovidVaccineRecord from "@/models/covidVaccineRecord";
+import { DateWrapper, StringISODate } from "@/models/dateWrapper";
+import { CustomBannerError, ResultError } from "@/models/errors";
+import { LoadStatus } from "@/models/storeOperations";
+import VaccinationStatus from "@/models/vaccinationStatus";
+import { ILogger, IVaccinationStatusService } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+export const useVaccinationStatusStore = defineStore(
+    "vaccinationStatusPublic",
+    () => {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+        const vaccinationStatusService =
+            container.get<IVaccinationStatusService>(
+                SERVICE_IDENTIFIER.VaccinationStatusService
+            );
+
+        const errorStore = useErrorStore();
+
+        interface PublicVaccinationStatus {
+            vaccinationStatus?: VaccinationStatus;
```

Comment:
Suggest renaming `vaccinationStatus` property to `data` to match other stores.

Reply:
Changed.